File: tests/parsers/samples/code/dart/dynamic.dart
import 'simple.dart';

typedef SummaryAction = String Function();

final Map<String, SummaryAction> registry = <String, SummaryAction>{
  'summarize': summarize,
};

String runDynamic([String name = 'summarize']) {
  final action = registry[name];
  if (action == null) {
    return 'unknown action';
  }
  return 'dynamic:${action()}';
}

File: tests/parsers/samples/code/lua/dynamic.lua
local simple = require("simple")

local dynamic = {}

local registry = {
    summarize = simple.summarize
}

function dynamic.run(name)
    name = name or "summarize"
    local action = registry[name]
    if not action then
        return "unknown"
    end
    return "dynamic:" .. action()
end

return dynamic
